worker/db.go

```diff
@@ -0,0 +1,48 @@
+package worker
+
+import (
+	"fmt"
+	"os"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"github.com/mini-maxit/worker/utils"
+)
+
+// Connect to the database using GORM
+func connectToDatabase() *gorm.DB {
+	DATABASE_USER := os.Getenv("DATABASE_USER")
+	DATABASE_PASSWORD := os.Getenv("DATABASE_PASSWORD")
+	DATABASE_NAME := os.Getenv("DATABASE_NAME")
+	DATABASE_SSL_MODE := os.Getenv("DATABASE_SSL_MODE")
+
+	dsn := fmt.Sprintf("host=localhost user=%s password=%s dbname=%s sslmode=%s",
+		DATABASE_USER, DATABASE_PASSWORD, DATABASE_NAME, DATABASE_SSL_MODE)
+
+	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{})
+	utils.FailOnError(err, "failed to connect to database")
+
+	// Migrate the schema if needed
+	err = db.AutoMigrate(&Solution{}, &SolutionResult{})
```

Comment:
Let's not migrate table definitions from worker. I would like to leave it for web component